cmd/draino/draino.go

```diff
@@ -142,33 +147,38 @@ func main() {
 		}
 	}
 
-	sf := cache.FilteringResourceEventHandler{FilterFunc: kubernetes.NodeSchedulableFilter, Handler: h}
-	cf := cache.FilteringResourceEventHandler{FilterFunc: kubernetes.NewNodeConditionFilter(*conditions), Handler: sf}
+	cf := cache.FilteringResourceEventHandler{FilterFunc: kubernetes.NewNodeConditionFilter(*conditions), Handler: h}
 	lf := cache.FilteringResourceEventHandler{FilterFunc: kubernetes.NewNodeLabelFilter(*nodeLabels), Handler: cf}
 	nodes := kubernetes.NewNodeWatch(cs, lf)
 
 	id, err := os.Hostname()
 	kingpin.FatalIfError(err, "cannot get hostname")
 
+	// use a Go context so we can tell the leaderelection code when we
+	// want to step down
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	lock, err := resourcelock.New(
 		resourcelock.EndpointsResourceLock,
 		*namespace,
 		kubernetes.Component,
 		cs.CoreV1(),
+		cs.CoordinationV1(),
 		resourcelock.ResourceLockConfig{
 			Identity:      id,
 			EventRecorder: kubernetes.NewEventRecorder(cs),
 		},
 	)
 	kingpin.FatalIfError(err, "cannot create lock")
 
-	leaderelection.RunOrDie(leaderelection.LeaderElectionConfig{
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
 		Lock:          lock,
-		LeaseDuration: 15 * time.Second,
-		RenewDeadline: 10 * time.Second,
-		RetryPeriod:   2 * time.Second,
+		LeaseDuration: 60 * time.Second,
```

Comment:
A bit of a drive by change.  Can you briefly comment on this? The original settings were introduced in https://github.com/planetlabs/draino/pull/56.

Reply:
The value seems to me a bit aggressive compared to the activity of the controller itself. Not having `draino` running for one minute does not have any real impact on the system. I mean we usually have grace-period, headroom and drainBuffer that are set to couple of minutes...

I can go back to pervious values and even make these value parameters, if you want.

Reply:
2535715fa22eaae7dab3059b1ab83853e90b6016